dropbox/dropbox.py

```diff
@@ -479,6 +521,39 @@ def _save_body_to_file(self, download_path, http_resp, chunksize=2**16):
                 for c in http_resp.iter_content(chunksize):
                     f.write(c)
 
+    def with_path_root(self, mode, *args, **kwargs):
+        """
+        Creates a clone of the Dropbox instance with the Dropbox-API-Path-Root header
+        as the appropriate serialized instance of PathRoot.
+
+        For more information, see
+        https://www.dropbox.com/developers/reference/namespace-guide#pathrootmodes
+
+        :param str mode: path root mode to use. Must be one of "home", "root", or "namespace_id"
```

Comment:
Stupid question: Why not make these constants (e.g., integer constants)? Are they values that are passed over-the-wire/documented in the API?

Reply:
I thought about it, but then I generalized the implementation below such that the `mode` parameter is simply passed through to `getattr`.  This way we would only need to update the doc string if a new path root mode is added.

Reply:
Now I'm beginning to second guess myself and debating if I should go with your approach of having the user create the PathRoot object and just pass it in as a param.

Reply:
I think either is defensible. Whatever you land on is fine.

Reply:
Since the user already has identify and pass in a `mode` string, might as well just let them instantiate the `PathRoot` they want.  It's most explicit of the three options (string, constant, PathRoot object) and the effort for all options on the part of the developer is about the same.  Stay tuned, will update the commit with this approach (and update the README with instructions).